geode-core/src/main/java/org/apache/geode/internal/statistics/HostStatHelper.java

```diff
@@ -17,154 +17,65 @@
 import java.net.InetAddress;
 import java.net.UnknownHostException;
 
-import org.apache.geode.InternalGemFireException;
 import org.apache.geode.Statistics;
-import org.apache.geode.internal.PureJavaMode;
 import org.apache.geode.internal.net.SocketCreator;
 import org.apache.geode.internal.statistics.platform.LinuxProcFsStatistics;
 import org.apache.geode.internal.statistics.platform.LinuxProcessStats;
 import org.apache.geode.internal.statistics.platform.LinuxSystemStats;
-import org.apache.geode.internal.statistics.platform.OSXProcessStats;
-import org.apache.geode.internal.statistics.platform.OSXSystemStats;
 import org.apache.geode.internal.statistics.platform.OsStatisticsFactory;
 import org.apache.geode.internal.statistics.platform.ProcessStats;
-import org.apache.geode.internal.statistics.platform.SolarisProcessStats;
-import org.apache.geode.internal.statistics.platform.SolarisSystemStats;
-import org.apache.geode.internal.statistics.platform.WindowsProcessStats;
-import org.apache.geode.internal.statistics.platform.WindowsSystemStats;
 
 /**
- * Provides native methods which fetch operating system statistics.
+ * Provides static methods which fetch operating system statistics.
+ * Only Linux OS is allowed.
  */
 public class HostStatHelper {
-  private static final int SOLARIS_CODE = 1; // Sparc Solaris
-  private static final int WINDOWS_CODE = 2;
-  private static final int LINUX_CODE = 3; // x86 Linux
-  private static final int OSX_CODE = 4; // Mac OS X
-
   private static final int PROCESS_STAT_FLAG = 1;
   private static final int SYSTEM_STAT_FLAG = 2;
 
-  private static final int osCode;
-
   static {
-    String osName = System.getProperty("os.name", "unknown");
-    if (!PureJavaMode.osStatsAreAvailable()) {
-      throw new RuntimeException("HostStatHelper not allowed in pure java mode");
-    } else if (osName.equals("SunOS")) {
-      osCode = SOLARIS_CODE;
-    } else if (osName.startsWith("Windows")) {
-      osCode = WINDOWS_CODE;
-    } else if (osName.startsWith("Linux")) {
-      osCode = LINUX_CODE;
-    } else if (osName.equals("Mac OS X")) {
-      osCode = OSX_CODE;
-    } else {
-      throw new InternalGemFireException(
-          String.format(
-              "Unsupported OS %s. Supported OSs are: SunOS(sparc Solaris), Linux(x86) and Windows.",
-              osName));
-    }
-  }
-
-  public static boolean isWindows() {
-    return osCode == WINDOWS_CODE;
-  }
-
-  public static boolean isUnix() {
-    return osCode != WINDOWS_CODE;
-  }
-
-  public static boolean isSolaris() {
-    return osCode == SOLARIS_CODE;
+    new OSVerifier();
   }
 
   public static boolean isLinux() {
```

Comment:
Is there still a need for a method that always returns true?

Reply:
I have changed the method in the last commit